src/portfolio.rs

```diff
@@ -116,26 +119,172 @@ impl Portfolio {
+            Cell::new("Price").add_attribute(Attribute::Bold),
+            Cell::new("Value").add_attribute(Attribute::Bold),
+            Cell::new("PnL").add_attribute(Attribute::Bold),
+            Cell::new("%Hist").add_attribute(Attribute::Bold),
+            Cell::new("%Day").add_attribute(Attribute::Bold),
+        ]);
+
+        let mut total_value = 0.0_f64;
+        let mut total_invested = 0.0_f64;
+        let mut total_pnl = 0.0_f64;
+
+        // Track previous total value for computing portfolio daily change
+        let mut total_prev_value_for_day = 0.0_f64;
         for position in &self.positions {
-            println!(
-                "{0: >26} | {1: >12} | {2: >10.2} | {3: >10.2}",
-                position.get_name(),
-                position.get_asset_class(),
-                position.get_amount(),
-                position.get_balance()
-            );
+            let amount = position.get_amount();
+            let avg_cost = position.average_cost();
+            let invested = position.total_invested();
+            let price = position.market_price();
+            let value = position.market_value();
+            let pnl = position.pnl();
+            let day_var = position.daily_variation_percent();
+            let hist_var = position.historic_variation_percent();
+
+            // Check if this is a cash position (no ticker and cash asset class)
+            let is_cash = position.get_ticker().is_none() && 
+                         position.get_asset_class().to_lowercase() == "cash";
+
+            total_value += value;
+            if let Some(i) = invested { total_invested += i; }
+            if let Some(p) = pnl { total_pnl += p; }
+
+            // Estimate previous value from daily % to compute portfolio daily change
+            let prev_value_for_row = match day_var {
+                Some(dv) => {
+                    let ratio = dv / 100.0;
+                    if (1.0 + ratio).abs() > f64::EPSILON {
+                        value / (1.0 + ratio)
+                    } else {
+                        value
+                    }
+                }
+                None => value,
+            };
+            total_prev_value_for_day += prev_value_for_row;
+
+            let avg_cost_str = if is_cash {
+                "-".to_string()
+            } else {
+                avg_cost.map(|v| format!("{v:.2}")).unwrap_or_else(|| "-".to_string())
+            };
+            let invested_str = if is_cash {
+                "-".to_string()
+            } else {
+                invested.map(|v| format!("{v:.2}")).unwrap_or_else(|| "-".to_string())
+            };
+            let asset_class = position.get_asset_class();
+            let asset_color = match asset_class.to_lowercase().as_str() {
+                "crypto" => TColor::DarkYellow,
+                "stocks" | "stock" => TColor::DarkBlue,
+                "cash" => TColor::DarkGreen,
+                _ => TColor::White,
```

Comment:
Nice thing you added colors, but it would be nice to add colors for all supported asset classes:

Stocks, Bonds, Commodities, Gold, Crypto, Cash